notebooks/224-stable-diffusion-text-to-image/224-stable-diffusion-text-to-image.ipynb

```diff
@@ -0,0 +1,1272 @@
+    "VAE_ONNX_PATH = Path('vae_decoder.onnx')\n",
+    "VAE_OV_PATH = VAE_ONNX_PATH.with_suffix('.xml')\n",
+    "\n",
+    "\n",
+    "def convert_vae(pipe:StableDiffusionPipeline, onnx_path:Path, ov_path:Path):\n",
+    "    \"\"\"\n",
+    "    Convert VAE model to ONNX, then IR format. \n",
+    "    Function accepts pipeline, creates wrapper class for export only necessary for inference part, \n",
+    "    prepares example inputs for ONNX conversion via torch.export, \n",
+    "    then convert using mo.convert and serialize model for saving on drive.\n",
+    "    Parameters: \n",
+    "        pipe (StableDiffusionPipeline): Stabloe Diffusion pipeline\n",
+    "        onnx_path (Path): File for storing onnx model\n",
+    "        ov_path (Path): File for IR serialization\n",
+    "    Returns:\n",
+    "        None\n",
+    "    \"\"\"\n",
+    "    class VAEDecoderWrapper(torch.nn.Module):\n",
+    "        def __init__(self, vae):\n",
+    "            super().__init__()\n",
+    "            self.vae = vae\n",
+    "\n",
+    "        def forward(self, latents):\n",
+    "            return self.vae.decode(latents)\n",
+    "\n",
+    "    if not onnx_path.exists():\n",
+    "        vae_decoder = VAEDecoderWrapper(pipe.vae)\n",
+    "        text = 'a photo of an astronaut riding a horse on mars'\n",
+    "        text_encoder = pipe.text_encoder\n",
+    "        input_ids = pipe.tokenizer(\n",
+    "            text,\n",
+    "            padding=\"max_length\",\n",
+    "            max_length=pipe.tokenizer.model_max_length,\n",
+    "            truncation=True,\n",
+    "            return_tensors=\"pt\",\n",
+    "        ).input_ids\n",
+    "        with torch.no_grad():\n",
+    "            text_encoder_output = text_encoder(input_ids)\n",
+    "        latents_shape = (2, 4, 512 // 8, 512 // 8)\n",
+    "        latents = torch.randn(latents_shape)\n",
+    "        t = torch.from_numpy(np.array(1, dtype=float))\n",
+    "        max_length = input_ids.shape[-1]\n",
+    "        uncond_input = pipe.tokenizer([\"\"], padding=\"max_length\", max_length=max_length, return_tensors=\"pt\")\n",
+    "        uncond_embeddings = pipe.text_encoder(uncond_input.input_ids)[0]\n",
+    "        encoder_hidden_state = torch.cat([uncond_embeddings, text_encoder_output[0]])\n",
+    "        output_latents = pipe.unet(latents, t, encoder_hidden_state)[0]\n",
+    "        latents_uncond, latents_text = output_latents[0].unsqueeze(0), output_latents[1].unsqueeze(0)\n",
+    "        latents = latents_uncond + 7.5 * (latents_text - latents_uncond)\n",
+    "\n",
+    "        vae_decoder.eval()\n",
+    "        with torch.no_grad():\n",
+    "            torch.onnx.export(vae_decoder, latents, onnx_path, input_names=['latents'], output_names=['sample'])\n",
+    "        print('VAE decoder successfully converted to ONNX')\n",
+    "    vae = mo.convert(input_model=onnx_path, data_type='FP16')  # convert model to OpenVINO model\n",
```

Comment:
In the recent master it is called `convert_model` and `input_model` can be specified as a positional argument.
Also as https://github.com/openvinotoolkit/openvino/pull/13832 has been merged as well, you can specify `compress_to_fp16` instead of `data_type=FP16`.